### DISCUSSION TITLE
Exporting jest config with next/jest yields issue with async code

### DISCUSSION THREAD

**Discussion by lorenzo-dallamuta** [vercel/next.js#60873] [Help] [answered]
### Summary

I'm facing an issue exporting Jest configurations in a Next project. I get an error that signifies I'm trying to export an undefined value from jest.config.js. The export is the same as with the docs and the examples, where the export line is commented as:

    createJestConfig is exported this way to ensure that next/jest can load the Next.js config which is async

It seems the async code is not being properly handled in my project, **what settings should I check at the project level to make sure this export can work?**

### Additional information

This is my jest.config.js:
```js
const nextJest = require("next/jest")
const createJestConfig = nextJest({dir: "./"})
const config = {
    coverageProvider: "babel",
    preset: "ts-jest",
    testEnvironment: "jest-environment-jsdom",
}
module.exports = createJestConfig(config)
```

This is the error I get:
```js
TypeError: Cannot read properties of undefined (reading '1')
at module.exports (/home/ldm-acn/gedi_drupal/gedi-fe-rsc/next.config.js:16:39)
```

This is my next.config.js:
```js
/** @type {import('next').NextConfig} */
module.exports = async (phase, { defaultConfig }) => {
    /**
     *@type {import('next').NextConfig}
     */
    const r = new RegExp(/https*:\/\/([\d\.]+):(\d+)\/*/g)
    const imageRemotePattern = Array.from(
        process.env.NEXT_PUBLIC_BASE_URL.matchAll(r),
    )[0]
    const nextConfig = {
        images: {
            remotePatterns: [
                {
                    protocol: "http",
                    hostname: imageRemotePattern[1],
                    port: imageRemotePattern[2],
                    pathname: "/**",
                },
            ],
        },
    }
    return nextConfig
}
```

**Comment by icyJoseph** [comment]
I think the error is happening when trying to read the `next.config.js` file... could you should what's going on there?

**Reply by lorenzo-dallamuta** [reply]
I added the next.config.js code to the question

**Reply by icyJoseph** [reply, marked as answer]
I think `imageRemorePattern` is empty, could you add a `console.log({imageRemotePattern})` before you declare `const nextConfig`?

**Reply by lorenzo-dallamuta** [reply]
You're right, `imageRemotePattern` is empty when I run the tests and if I take out the code that reads from `process.env` the tests can run. If this file is called before the testing environment can gain access to the environment variables it makes sense the next/jest function doesn't work 😛

**Reply by lorenzo-dallamuta** [reply]
I guess I'll have to find a different way to set image.remotePatterns when I need to run a test

**Reply by icyJoseph** [reply]
Create, `env.test` and put the content you need there.

**Reply by lorenzo-dallamuta** [reply]
for some reason that variable didn't exist in my `env.test`, I must have copied from a different branch... now it works
